src/tools/PrizeTierController/utils/checkForPrizeCompatibilityErrors.ts

```diff
@@ -0,0 +1,129 @@
+import { PrizeTierConfig } from '@pooltogether/v4-utils-js'
+import { ethers } from 'ethers'
+
+interface PrizePoolCompatibilityErrors {
+  chainId: number
+  address: string
+  errors: ('prizeAmount' | 'maxPicks' | 'bitRange')[]
+}
+
+interface SimplePrizePoolConfig {
+  chainId: number
+  address: string
+  prizeAmount: number
+  maxPicks: number
+  bitRange: number
+}
+
+/**
+ * Checks for any compatibility errors between pool configs.
+ * @param prizePoolConfigs - The configurations of all prize pools.
+ * @returns Array of errored prize pools.
+ */
+export const checkForPrizeCompatibilityErrors = (prizePoolConfigs: {
+  [chainId: number]: { [prizeTierHistoryAddress: string]: PrizeTierConfig }
+}): PrizePoolCompatibilityErrors[] => {
+  const compatibilityErrors: PrizePoolCompatibilityErrors[] = []
+  const prizePoolConfigsArray: SimplePrizePoolConfig[] = []
+  const prizeAmounts: number[] = []
+  const maxPicks: number[] = []
+  const bitRanges: number[] = []
+
+  // Fetching values from each prize config:
+  Object.keys(prizePoolConfigs).forEach((chainId) => {
```

Comment:
`Object.keys` returns results as strings. As you've noticed we usually use `chainId` as a key, which is a number. Safest thing to do is `Object.keys(prizePoolConfigs).map(Number)` (parseInt works too, I've just been using Number to be consistent) to immediately convert it. One day we'll clean that all up so it doesn't cause bugs but for now lets just keep that consistent so it's easy to upgrade in the future.

Reply:
This is old code from when checking if prize pools were in sync was a thing - I've gone ahead and just removed it altogether in f5cbb5d